modules/serverless-kafka-sasl-source.adoc

```diff
@@ -0,0 +1,127 @@
+* You have a username and password for a Kafka cluster.
+* You have chosen the SASL mechanism to use, for example `PLAIN`, `SCRAM-SHA-256`, or `SCRAM-SHA-512`.
+* If TLS is enabled, you also need the `ca.crt` certificate file for the Kafka cluster.
+* Install the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Create the certificate files as secrets in your chosen namespace:
++
+[source,terminal]
+----
+$ oc create secret -n <namespace> generic <kafka_auth_secret> \
+  --from-file=ca.crt=caroot.pem \
+  --from-literal=password="SecretPassword" \
+  --from-literal=saslType="SCRAM-SHA-512" \
+  --from-literal=user="my-sasl-user"
+----
+** Use the key names `ca.crt`, `password`, and `sasl.mechanism`. Do not change them.
+** If you want to use SASL with public CA certificates, you must use the `tls.enabled=true` flag, rather than the `ca.crt` argument, when creating the secret. For example:
++
+[source,terminal]
+----
+$ oc create secret -n <namespace> generic <kafka_auth_secret> \
+  --from-literal=tls.enabled=true \
+  --from-literal=password="SecretPassword" \
+  --from-literal=saslType="SCRAM-SHA-512" \
+  --from-literal=user="my-sasl-user"
+----
+
+. Start editing the `KnativeKafka` custom resource:
++
+[source,terminal]
+----
+$ oc edit knativekafka
+----
+
+. Reference your secret and the namespace of the secret:
++
+[source,yaml]
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  channel:
+    authSecretName: <kafka_auth_secret>
+    authSecretNamespace: <kafka_auth_secret_namespace>
+    bootstrapServers: <bootstrap_servers>
+    enabled: true
+  source:
+    enabled: true
+----
++
+[NOTE]
+====
+Make sure to specify the matching port in the bootstrap server.
+====
++
+For example:
++
+[source,yaml]
+----
+apiVersion: operator.serverless.openshift.io/v1alpha1
+kind: KnativeKafka
+metadata:
+  namespace: knative-eventing
+  name: knative-kafka
+spec:
+  channel:
+    authSecretName: scram-user
+    authSecretNamespace: kafka
+    bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9093
+    enabled: true
+  source:
+    enabled: true
+----
+
+. Create or modify your Kafka source so that it contains the following `spec` configuration:
++
+[source,yaml]
+----
+apiVersion: sources.knative.dev/v1beta1
+kind: KafkaSource
+metadata:
+  name: example-source
+spec:
+...
+  net:
+    sasl:
+      enable: true
+      user:
+        secretKeyRef:
+          name: <secret-name>
+          key: user
+      password:
+        secretKeyRef:
+          name: <secret-name>
+          key: password
+      saslType:
+        secretKeyRef:
+          name: <secret-name>
+          key: saslType
+    tls:
+      enable: true
+      caCert:
+        secretKeyRef:
+          name: <secret-name>
+          key: ca.crt
```

Comment:
Since we use the `<kafka_auth_secret>` placeholder for the command `oc create secret -n <namespace> generic <kafka_auth_secret>` above, for consistency we should use the same here.